### PR TITLE
feat: break and continue conditions

### PR DESCRIPTION
**Pull request by TAdev0**
@enitrat  i think my example is not ideal because i need to add a sub operation in the printing for each iteration to print the same result...

- - -
This change is [Reviewable](https://reviewable.io/reviews/cairo-book/cairo-book/846)

### CODE REVIEW

File: src/ch02-05-control-flow.md [hunk 1]
```diff
@@ -149,6 +149,14 @@ Let's add a `continue` statement to our loop to skip the `println!` statement wh
 
 Executing this program will not print the value of `i` when it is equal to `5`.
 
+> When defining `break` or `continue` conditions in loops, it is better design to avoid comparison operators. Indeed, it is much cheaper to check if `i == 11` instead of `i > 10`, because comparison operators involve additional Cairo steps related to `range_check` Cairo-VM builtin usage.
```
Comment (TAdev0) [inline src/ch02-05-control-flow.md:152]: i know its brutal in chapter 2 to talk about range check builtin. but later we can add a reference here for the cairo vm book section about builtins. Maybe i should add right now a ''future'' reference?

**Review by enitrat — COMMENTED** [review]
I'm not sure what was the intention but this is already covered in the control flow chapter and I don't see a value added

**Comment by TAdev0** [comment]
@enitrat  where is it covered?

it aims to resolve this issue : 

https://github.com/cairo-book/cairo-book/issues/463

just added a short section explaining why its better design to compare equality and not comparison for loops

**Comment by TAdev0** [comment]
the value added is that by reading this, people might start using == for looping break and continue, thus saving useless computation

**Comment by enitrat** [comment]
Will close this as I don't want to bloat the book with small nits about gas consumption. Eventually it will be meaningless

**Comment by TAdev0** [comment]
makes sense, now i finally agree with you

**Pull request closed without merging**